mautic/mautic
# Dashboard widget with segments build time

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds a new dashboard widget that displays a list of the segments with the longest rebuild time. This allows you to monitor which segments have the most time-consuming filters.
![image](https://github.com/mautic/mautic/assets/8580942/66882278-8568-437c-8045-bb01b7f13334)





#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add some segments with filters
3. Run segments update job `ddev exec bin/console m:s:u`
4. Add Segments build time widget to dashboard
![image](https://github.com/mautic/mautic/assets/8580942/fe86d0f5-9306-4fd2-aeee-0e9a34023554)
5. Widget should be visible in the dashboard
